### DISCUSSION TITLE
Deploying your Azure virtual machine

### DISCUSSION THREAD

**Discussion by EY-glowglowglow** [EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge#41] [3.0 VMs and data (Azure VM, local deployment, and Jupyter)]
Please refer [here ](https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/1b.-Getting-started:-Azure-Environment) for instructions, and comment below if you get any errors with a screenshot of your error.

**Comment by zaqaus** [comment]
## Unable to access jupyter notebook

I am unble to connect to the jupyter notebook using the public IP address of my VM.
Following the instruction on [4a. Troubleshooting: FAQ],(https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/4a.-Troubleshooting:-FAQ#access-the-vm-terminal) I was able to access my VM using Windows command prompt.

However, when I run `cd /opt/odc` and `sudo docker-compose ps`, I get an error:
> ERROR: Invalid interpolation format for "command" option in service "jupyter": "jupyter notebook --allow-root --ip="0.0.0.0" --NotebookApp.token='my-password'" (see the screenshot below)

I need help!
![ey-1](https://user-images.githubusercontent.com/82909733/115465691-515db800-a272-11eb-96a2-b9ffe44106c1.png)

Thanks

**Reply by alexgleith** [reply]
What does your `docker-compose.yml` file look like?

Specifically, this line: https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/blob/main/docker-compose.yml#L32

**Reply by zaqaus** [reply]
@alexgleith Thanks for your reply.
I was able to resolve the issue by deleting the resource group and creating a new one.

**Comment by alvinlui49** [comment]
Hi, I get the following error when I click on the template

The template deployment failed with error: 'The resource with id: '/subscriptions/****************/resourceGroups/EY_DataScience_2021/providers/Microsoft.Compute/virtualMachines/CubeInABox-CPU-16GB' failed validation with message: 'The requested size for resource '/subscriptions/****************/resourceGroups/EY_DataScience_2021/providers/Microsoft.Compute/virtualMachines/CubeInABox-CPU-16GB' is currently not available in location 'australiaeast' zones '' for subscription '****************'. Please try another size or deploy to a different location or zones. See https://aka.ms/azureskunotavailable for details.'.'. (Code: InvalidTemplateDeployment)

<img width="769" alt="image" src="https://user-images.githubusercontent.com/83225628/116090402-67d29c00-a6d6-11eb-93aa-8e6cadb18748.png">

**Reply by EY-glowglowglow** [reply]
Hi there,
This could be related to a quota limit already reached. 
If you are using Azure for Students, you will not be able to increase the quota. You would need to deallocate existing VMs to deploy new ones. 
If you're using a different subscription (not Azure for Students) you can increase the quota following instructions [here ](https://docs.microsoft.com/en-us/azure/azure-portal/supportability/per-vm-quota-requests)

**Comment by konsweb-122** [comment]
I have copied my public IP address  when running the jupyter VM to a new tab but i still cant access. What should I do? 

I had sucessfully run all the follow code in my Windows command prompt (at https://github.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/wiki/4a.-Troubleshooting:-FAQ#cannot-access-jupyter): 
""""
ssh -i /path/to/key ubuntu@VM_IP
cd /opt/odc 
sudo docker-compose ps 
sudo docker-compose -d up 

cd # Navigate back to your home directory
curl -o install-cube.sh https://raw.githubusercontent.com/EY-Data-Science-Program/2021-Better-Working-World-Data-Challenge/main/install-cube.sh

chmod +x install-cube.sh 
sudo ./install-cube.sh 1234
""""

**Reply by EY-glowglowglow** [reply]
Hi @codeindulgence any ideas on this error?

**Reply by nickbutler** [reply]
Hi @konsweb-122, 

Can you share the output of the `ps` command:
```
cd /opt/odc
sudo docker-compose ps
```
